sql-plugin/src/main/scala/com/nvidia/spark/rapids/Plugin.scala

```diff
@@ -86,22 +94,54 @@ object RapidsPluginUtils extends Logging {
-          logWarning("Rapids SQL Plugin when used with Kryo needs to register some " +
-            s"serializers using $KRYO_REGISRATOR_NAME. Please call it from your registrator " +
-            " to let the plugin work properly.")
+      if (conf.contains(KRYO_REGISTRATOR_KEY)) {
+        if (!KRYO_REGISTRATOR_NAME.equals(conf.get(KRYO_REGISTRATOR_KEY)) ) {
+          logWarning("The RAPIDS Accelerator when used with Kryo needs to register some " +
+              s"serializers using $KRYO_REGISTRATOR_NAME. Please call it from your registrator " +
+              " to let the plugin work properly.")
         } // else it is set and we are good to go
       }  else {
         // We cannot set the kryo key here, it is not early enough to be picked up everywhere
-        throw new UnsupportedOperationException("The Rapids SQL Plugin when used with Kryo needs " +
-          s"to register some serializers. Please set the spark config $KRYO_REGISRATOR_KEY to " +
-          s"$KRYO_REGISRATOR_NAME or some operations may not work properly.")
+        throw new UnsupportedOperationException("The RAPIDS Accelerator when used with Kryo " +
+            "needs to register some serializers. Please set the spark config " +
+            s"$KRYO_REGISTRATOR_KEY to $KRYO_REGISTRATOR_NAME or some operations may not work " +
+            "properly.")
       }
     } else if (!JAVA_SERIALIZER_NAME.equals(serializer)) {
       throw new UnsupportedOperationException(s"$serializer is not a supported serializer for " +
-        s"the Rapids SQL Plugin. Please disable the rapids plugin or use a supported serializer " +
-        s"serializer ($JAVA_SERIALIZER_NAME, $KRYO_SERIALIZER_NAME).")
+          s"the RAPIDS Accelerator. Please disable the RAPIDS Accelerator or use a supported " +
+          s"serializer ($JAVA_SERIALIZER_NAME, $KRYO_SERIALIZER_NAME).")
+    }
+  }
+
+  /** Gets the plugin version string from the plugin properties file in the classpath */
+  def getPluginVersion: String = {
+    getVersionFromPropsFile(PLUGIN_PROPS_FILENAME)
+  }
+
+  /** Gets the cudf version string from the cudf properties file in the classpath*/
+  def getClasspathCudfVersion: String = {
+    getVersionFromPropsFile(CUDF_PROPS_FILENAME)
+  }
+
+  /** Loads the plugin properties file in the classpath */
+  def getPluginProps: Properties = loadProps(PLUGIN_PROPS_FILENAME)
+
+  private def loadProps(resourceName: String): Properties = {
+    val classLoader = RapidsPluginUtils.getClass.getClassLoader
+    val resource = classLoader.getResourceAsStream(resourceName)
+    if (resource == null) {
+      throw new PluginException(s"Could not find properties file $resourceName in the classpath")
+    }
+    val props = new Properties
+    props.load(resource)
+    props
+  }
+
+  private def getVersionFromPropsFile(resourceName: String): String = {
+    val props = loadProps(resourceName)
+    Option(props.get("version")).map(_.toString).getOrElse {
```

Comment:
Is this going to get called from the driver too? If so, I wonder if we should have an "undetected" state, in case cudf isn't in the class path of the driver.

Reply:
cudf has to be in the classpath of the driver.  The driver calls some cudf Java code that doesn't load libcudf.

Reply:
Cool that makes sense